src/dapf.c

```diff
@@ -110,28 +117,44 @@ int dapf_init(const char *path)
         return -1;
     }
 
+    if (pwr)
+        pmgr_adt_power_disable(path);
+
     if (!ret)
         printf("dapf: Initialized %s\n", path);
 
     return ret;
 }
 
-const char *dapf_paths[] = {"/arm-io/dart-aop", "/arm-io/dart-mtp", "/arm-io/dart-pmp", NULL};
+struct entry {
+    const char *path;
+    int index;
+};
+
+struct entry dapf_entries[] = {
+    {"/arm-io/dart-aop", 1},
+    {"/arm-io/dart-mtp", 1},
+    {"/arm-io/dart-pmp", 1},
+    {"/arm-io/dart-isp", 5},
```

Comment:
We should consider deriving the register range index from ADT. I think it was the `instance` property that held that info? Would that work here?

Reply:
`instance` are names of each dapf entry (DAPF, SMMU, TLL, DART). I couldn't find another prop either. But for all of my test cases (t8103 ane/aop/isp; mtp/pmp gets dropped anyway) it's the last range index, which we could get from `reg_len / ((a_cells + s_cells) * 4)`. If this holds up I can revert this & add `adt_get_reg_count()` instead.